gap/semigroups/semifp.gi

```diff
@@ -43,6 +43,20 @@ function(x)
   return ExtRepOfObj(UnderlyingElement(x));
 end);
 
+InstallMethod(ElementOfFpSemigroup,
+"for an fp semigroup and an associative word",
+[IsFpSemigroup, IsAssocWord],
+function(s, w)
+  return ElementOfFpSemigroup(FamilyObj(s.1), w);
```

Comment:
Might it not be a good idea to add a check that `S` has at least one generator here? I think it's not possible to create an fp semigroup with 0 generators, but it'd be good to check/assert this explicitly here, maybe?

Reply:
I'm happy to implement this, hadn't really thought about that. I'm not sure how many ways there are of creating an fp semigroup with 0 generators; I've found this method, but surprisingly `s.1` is still defined and works fine.

```
gap> f := FreeSemigroup([]);
<free semigroup on the generators [  ]>
gap> s := f / [];
<fp semigroup on the generators [  ]>
gap> s.1;

gap> s.1 = ElementOfFpSemigroup(FamilyObj(s.1), f.1);
true
```

The issue will be exactly the same for `ElementOfFpMonoid` by the way, since the generators `m.1`, `m.2` etc. do not include the identity. What do you think we should do, and how should we check that `s.1` will be defined?